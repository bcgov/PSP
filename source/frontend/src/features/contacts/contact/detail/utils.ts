import { ContactInfoField } from 'features/contacts/interfaces';
import { Dictionary } from 'interfaces/Dictionary';
import { IContactAddress, IContactMethod } from 'interfaces/IContact';

import { AddressTypes } from './../../../../constants/addressTypes';

// the order of this array corresponds to the expected display order
const addressSortOrder = [AddressTypes.Mailing, AddressTypes.Residential, AddressTypes.Billing];

export const sortAddresses = (a1: IContactAddress, a2: IContactAddress) => {
  if (a2.addressType.id === a1.addressType.id) {
    return 0;
  }
  const a2Index = addressSortOrder.indexOf(a2.addressType.id as AddressTypes);
  if (a2Index !== -1 && a2Index < addressSortOrder.indexOf(a1.addressType.id as AddressTypes)) {
    return 1;
  }

  return -1;
};

export function getContactInfo(
  validTypes: Dictionary<string>,
  contactMethods?: IContactMethod[],
): ContactInfoField[] {
  if (contactMethods === undefined) {
    return [];
  }
  // Get only the valid types
  let filteredFields = contactMethods.reduce(
    (accumulator: ContactInfoField[], method: IContactMethod) => {
      if (Object.keys(validTypes).includes(method.contactMethodType.id)) {
        accumulator.push({
          info: method.value,
          label: validTypes[method.contactMethodType.id],
        });
      }
      return accumulator;
    },
    [],
  );

  // Sort according to the dictionary order
  return filteredFields.sort((a, b) => {
    return Object.values(validTypes).indexOf(a.label) - Object.values(validTypes).indexOf(b.label);
  });
}

export const fakeAddresses = [
  {
    id: 1,
    rowVersion: 3,
    addressType: {
      id: 'BILLING',
      description: 'Billing address',
      isDisabled: false,
    },
    streetAddress1: 'Billing address',
    municipality: 'Hollywood North',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V6A 5G7',
  },
  {
    id: 1,
    rowVersion: 3,
    addressType: {
      id: 'MAILING',
      description: 'Mailing address',
      isDisabled: false,
    },
    streetAddress1: 'Mailing address',
    streetAddress2: 'Living in a van',
    streetAddress3: 'Down by the River',
    municipality: 'Hollywood North',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V6Z 5G7',
  },
  {
    id: 5,
    rowVersion: 1,
    addressType: {
      id: 'RESIDNT',
      description: 'Property address',
      isDisabled: false,
    },
    streetAddress1: 'Property address',
    streetAddress2: '',
    streetAddress3: '',
    municipality: 'vic',
    province: {
      provinceStateId: 1,
      provinceStateCode: 'BC',
      description: 'British Columbia',
    },
    country: {
      countryId: 1,
      countryCode: 'CA',
      description: 'Canada',
    },
    postal: 'V9A 0H6',
  },
];
