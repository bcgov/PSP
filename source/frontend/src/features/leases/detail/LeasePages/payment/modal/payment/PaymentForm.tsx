import { Formik, FormikProps, validateYupSchema, yupToFormErrors } from 'formik';

import { ApiGen_CodeTypes_LeasePaymentCategoryTypes } from '@/models/api/generated/ApiGen_CodeTypes_LeasePaymentCategoryTypes';
import { ApiGen_Concepts_LeasePeriod } from '@/models/api/generated/ApiGen_Concepts_LeasePeriod';

import { FormLeasePayment, FormLeasePeriod } from '../../models';
import { isActualGstEligible } from '../../PeriodPaymentsContainer';
import PaymentFormContent from './PaymentFormContent';
import { PaymentsYupSchema } from './PaymentsYupSchema';

export interface IPaymentFormProps {
  formikRef: React.Ref<FormikProps<FormLeasePayment>>;
  onSave: (values: FormLeasePayment) => void;
  initialValues?: FormLeasePayment;
  isReceived?: boolean;
  periods: ApiGen_Concepts_LeasePeriod[];
}

/**
 * Internal Form intended to be displayed within a modal window. will be initialized with default values
 * if not initialValues provided. Otherwise will display the passed lease payment. Save/Cancel triggered externally via passed formikRef.
 * @param {IPaymentFormProps} props
 */
export const PaymentForm: React.FunctionComponent<React.PropsWithChildren<IPaymentFormProps>> = ({
  initialValues,
  formikRef,
  onSave,
  isReceived,
  periods,
}: IPaymentFormProps) => {
  let isGstEligible = false;
  if (initialValues?.leasePeriodId) {
    isGstEligible = isActualGstEligible(
      initialValues?.leasePeriodId,
      periods?.map(t => FormLeasePeriod.fromApi(t)) ?? [],
      ApiGen_CodeTypes_LeasePaymentCategoryTypes[initialValues?.leasePaymentCategoryTypeCode?.id],
    );
  }

  const currentPeriod = periods.find(t => t.id === initialValues?.leasePeriodId);

  return (
    <Formik<FormLeasePayment>
      innerRef={formikRef}
      enableReinitialize
      validateOnChange={false}
      validate={values => {
        let errors = {};
        try {
          validateYupSchema(values, PaymentsYupSchema, true);
        } catch (err) {
          errors = yupToFormErrors(err);
        }
        if (values.amountTotal !== +values.amountGst + +values.amountPreTax) {
          return {
            ...errors,
            form: 'Expected payment amount and GST amount must sum to the total received',
          };
        }

        return errors;
      }}
      onSubmit={values => {
        onSave(values);
      }}
      initialValues={{
        ...initialValues,
        amountGst: isGstEligible ? initialValues?.amountGst ?? '' : '',
      }}
    >
      <PaymentFormContent
        isReceived={!!isReceived}
        isGstEligible={!!isGstEligible}
        isVariable={currentPeriod?.isVariable}
      />
    </Formik>
  );
};

export default PaymentForm;
