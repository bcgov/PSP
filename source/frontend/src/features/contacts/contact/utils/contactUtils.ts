import { CountryCodes } from 'constants/index';
import { AddressField } from 'features/contacts/interfaces';
import { validateYupSchema, yupToFormErrors } from 'formik';
import { IEditableOrganizationForm, IEditablePersonForm } from 'interfaces/editable-contact';
import isEmpty from 'lodash/isEmpty';

import {
  hasAddress,
  hasEmail,
  hasPhoneNumber,
  OrganizationValidationSchema,
  PersonValidationSchema,
} from '../create/validation';
import { sortAddresses } from '../detail/utils';
import { IContactAddress } from './../../../../interfaces/IContact';

export const onValidateOrganization = (
  values: IEditableOrganizationForm,
  otherCountryId?: string,
) => {
  const errors = {} as any;
  try {
    // combine yup schema validation with custom rules
    if (!hasEmail(values) && !hasPhoneNumber(values) && !hasAddress(values)) {
      errors.needsContactMethod =
        'Contacts must have a minimum of one method of contact to be saved. (ex: email,phone or address)';
    }
    validateYupSchema(values, OrganizationValidationSchema, true, {
      otherCountry: otherCountryId,
    });
    return errors;
  } catch (err) {
    return { ...errors, ...yupToFormErrors(err) };
  }
};

export const onValidatePerson = (values: IEditablePersonForm, otherCountryId?: string) => {
  const errors = {} as any;
  try {
    // combine yup schema validation with custom rules
    if (!hasEmail(values) && !hasPhoneNumber(values) && !hasAddress(values)) {
      errors.needsContactMethod =
        'Contacts must have a minimum of one method of contact to be saved. (ex: email,phone or address)';
    }
    validateYupSchema(values, PersonValidationSchema, true, { otherCountry: otherCountryId });

    return errors;
  } catch (err) {
    return { ...errors, ...yupToFormErrors(err) };
  }
};

export const toAddressFields = (addresses: IContactAddress[]) => {
  return addresses
    .sort(sortAddresses)
    .reduce((accumulator: AddressField[], value: IContactAddress) => {
      accumulator.push({
        label: value.addressType.description || '',
        streetAddress1: value.streetAddress1,
        streetAddress2: value.streetAddress2,
        streetAddress3: value.streetAddress3,
        municipalityAndProvince: [
          value.municipality !== undefined ? value.municipality : '',
          value.province?.provinceStateCode ?? '',
        ]
          .filter(x => !isEmpty(x))
          .join(' '),
        country:
          value.country?.countryCode === CountryCodes.Other
            ? value.countryOther ?? ''
            : value.country?.description,
        postal: value.postal,
      });
      return accumulator;
    }, []);
};
