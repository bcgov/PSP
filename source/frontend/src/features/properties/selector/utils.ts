import { IMapProperty } from 'features/properties/selector/models';
import { compact } from 'lodash';
import { Api_PropertyFile } from 'models/api/PropertyFile';
import { formatApiAddress, pidFormatter } from 'utils';

import { Api_Geometry, Api_Property } from './../../../models/api/Property';

export enum NameSourceType {
  PID = 'PID',
  PIN = 'PIN',
  PLAN = 'Plan #',
  LOCATION = 'Location',
  NAME = 'Descriptive Name',
  NONE = 'n/a',
  ADDRESS = 'Address',
}

interface PropertyName {
  label: NameSourceType;
  value: string;
}

export const getPropertyName = (property: IMapProperty): PropertyName => {
  if (property.pid !== undefined && property.pid?.toString().length > 0 && property.pid !== '0') {
    return { label: NameSourceType.PID, value: pidFormatter(property.pid.toString()) };
  } else if (
    property.pin !== undefined &&
    property.pin?.toString()?.length > 0 &&
    property.pin !== '0'
  ) {
    return { label: NameSourceType.PIN, value: property.pin.toString() };
  } else if (property.planNumber !== undefined) {
    return { label: NameSourceType.PLAN, value: property.planNumber };
  } else if (property.latitude !== undefined && property.longitude !== undefined) {
    return {
      label: NameSourceType.LOCATION,
      value: compact([property.longitude?.toFixed(6), property.latitude?.toFixed(6)]).join(', '),
    };
  } else if (property.address !== undefined) {
    return {
      label: NameSourceType.ADDRESS,
      value: property.address,
    };
  }
  return { label: NameSourceType.NONE, value: '' };
};

export const getPrettyLatLng = (location?: Api_Geometry) =>
  compact([
    location?.coordinate?.x?.toFixed(6) ?? 0,
    location?.coordinate?.y?.toFixed(6) ?? 0,
  ]).join(', ');

export const getFilePropertyName = (
  fileProperty?: Api_PropertyFile,
  skipName: boolean = false,
): PropertyName => {
  if (fileProperty === undefined) {
    return { label: NameSourceType.NONE, value: '' };
  }

  if (
    fileProperty.propertyName !== undefined &&
    fileProperty.propertyName !== '' &&
    skipName === false
  ) {
    return { label: NameSourceType.NAME, value: fileProperty.propertyName };
  } else if (fileProperty.property !== undefined) {
    const property = fileProperty.property;
    return getApiPropertyName(property);
  }
  return { label: NameSourceType.NONE, value: '' };
};

export const getApiPropertyName = (property?: Api_Property): PropertyName => {
  if (property === undefined) {
    return { label: NameSourceType.NONE, value: '' };
  }

  let mapProperty: IMapProperty = {
    pin: property.pin?.toString(),
    pid: property.pid?.toString(),
    latitude: property.latitude,
    longitude: property.longitude,
    planNumber: property.planNumber,
    address: property.address !== undefined ? formatApiAddress(property.address) : undefined,
  };
  return getPropertyName(mapProperty);
};
