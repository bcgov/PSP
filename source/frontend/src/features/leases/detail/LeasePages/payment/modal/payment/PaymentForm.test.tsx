import axios from 'axios';
import MockAdapter from 'axios-mock-adapter';
import { Formik } from 'formik';
import { createMemoryHistory } from 'history';
import noop from 'lodash/noop';

import { mockLookups } from '@/mocks/lookups.mock';
import { lookupCodesSlice } from '@/store/slices/lookupCodes';
import { systemConstantsSlice } from '@/store/slices/systemConstants';
import { act, fillInput, findByText, renderAsync, RenderOptions } from '@/utils/test-utils';

import {
  defaultFormLeasePayment,
  defaultFormLeasePeriod,
  FormLeasePayment,
  FormLeasePeriod,
} from '../../models';
import { isActualGstEligible as isActualGstEligibleOriginal } from '../../PeriodPaymentsContainer';
import PaymentForm, { IPaymentFormProps } from './PaymentForm';
import { ApiGen_Concepts_LeasePeriod } from '@/models/api/generated/ApiGen_Concepts_LeasePeriod';

const isActualGstEligible = vi.mocked(isActualGstEligibleOriginal);
const history = createMemoryHistory();
const mockAxios = new MockAdapter(axios);
const onSave = vi.fn();
const submitForm = vi.fn();
const storeState = {
  [lookupCodesSlice.name]: { lookupCodes: mockLookups },
  [systemConstantsSlice.name]: { systemConstants: [{ name: 'GST', value: '5.0' }] },
};
vi.mock('../../PeriodPaymentsContainer', () => ({
  isActualGstEligible: vi.fn(),
}));

describe('PaymentForm component', () => {
  const setup = async (
    renderOptions: RenderOptions &
      Partial<IPaymentFormProps> & {
        initialValues?: FormLeasePayment;
        periods?: ApiGen_Concepts_LeasePeriod[];
      } = {},
  ) => {
    // render component under test
    const component = await renderAsync(
      <Formik initialValues={renderOptions.initialValues ?? {}} onSubmit={noop}>
        <PaymentForm
          periods={renderOptions.periods ?? []}
          initialValues={renderOptions.initialValues}
          onSave={onSave}
          formikRef={{ current: { submitForm } } as any}
        />
      </Formik>,
      {
        ...renderOptions,
        store: renderOptions.store ?? storeState,
        history,
      },
    );

    return {
      component,
    };
  };

  beforeEach(() => {
    mockAxios.resetHistory();
    isActualGstEligible.mockReset();
  });
  it('renders as expected', async () => {
    const { component } = await setup({});

    expect(component.asFragment()).toMatchSnapshot();
  });
  it('renders with data as expected', async () => {
    const { component } = await setup({
      initialValues: { ...defaultFormLeasePayment },
    });

    expect(component.asFragment()).toMatchSnapshot();
  });

  it('Non-variable payments do not display payment category', async () => {
    isActualGstEligible.mockReturnValue(true);
    const {
      component: { container, queryByLabelText },
    } = await setup({
      initialValues: { ...defaultFormLeasePayment, leasePeriodId: 1 },
      periods: [{ ...FormLeasePeriod.toApi(defaultFormLeasePeriod), isVariable: false, id: 1 }],
    });

    const category = queryByLabelText('Payment category:');
    expect(category).toBeNull();
  });

  it('Variable payments do not display payment category', async () => {
    isActualGstEligible.mockReturnValue(true);
    const {
      component: { container, queryByLabelText },
    } = await setup({
      initialValues: { ...defaultFormLeasePayment, leasePeriodId: 1 },
      periods: [{ ...FormLeasePeriod.toApi(defaultFormLeasePeriod), isVariable: true, id: 1 }],
    });

    const category = queryByLabelText('Payment category:');
    expect(category).toBeVisible();
  });

  it('Entering a total amount results in a calculated gst amount and pre tax amount', async () => {
    isActualGstEligible.mockReturnValue(true);
    const {
      component: { container, findByLabelText },
    } = await setup({ initialValues: { ...defaultFormLeasePayment, leasePeriodId: 1 } });

    await act(async () => {
      await fillInput(container, 'amountTotal', '1050');
    });
    const amountPreTax = await findByLabelText('Expected payment ($)');
    const amountGst = await findByLabelText('GST ($)');
    expect(amountPreTax).toHaveValue('$1,000.00');
    expect(amountGst).toHaveValue('$50.00');
  });

  it('Entering a total amount results and a mismatched gst amount results in an error', async () => {
    isActualGstEligible.mockReturnValue(true);
    const {
      component: { container, findByText, findByDisplayValue },
    } = await setup({ initialValues: { ...defaultFormLeasePayment, leasePeriodId: 1 } });

    await act(async () => {
      await fillInput(container, 'amountGst', '50');
    });
    await findByDisplayValue('$50.00');
    const warningMessage = await findByText(
      'Expected payment amount and GST amount must sum to the total received',
    );
    expect(warningMessage).toBeVisible();
  });

  it('Entering a total amount results in a calculated pre tax amount when gst is not enabled', async () => {
    isActualGstEligible.mockReturnValue(false);
    const {
      component: { container, findByLabelText },
    } = await setup({ initialValues: { ...defaultFormLeasePayment, leasePeriodId: 1 } });

    await act(async () => {
      await fillInput(container, 'amountTotal', '1000');
    });
    const amountPreTax = await findByLabelText('Expected payment ($)');
    expect(amountPreTax).toHaveValue('$1,000.00');
  });

  it('Entering a total amount results in a calculated pre tax amount when gst constant is not available', async () => {
    isActualGstEligible.mockReturnValue(true);
    const {
      component: { container, findByLabelText },
    } = await setup({
      store: {
        ...storeState,
        systemConstant: {},
        initialValues: { ...defaultFormLeasePayment, leasePeriodId: 1 },
      },
    });

    await act(async () => {
      await fillInput(container, 'amountTotal', '1000');
    });
    const amountPreTax = await findByLabelText('Expected payment ($)');
    expect(amountPreTax).toHaveValue('$1,000.00');
  });
});
