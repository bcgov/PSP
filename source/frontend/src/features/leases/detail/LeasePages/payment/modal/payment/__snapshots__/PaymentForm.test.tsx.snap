// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`PaymentForm component > renders as expected 1`] = `
<DocumentFragment>
  <div
    class="Toastify"
  />
  <div />
  .c1 .react-datepicker__calendar-icon {
  width: 2.4rem;
  height: 3rem;
  margin-top: 0.5rem;
  margin-right: 1.2rem;
  right: 0;
  pointer-events: none;
}

.c1 .react-datepicker__view-calendar-icon input {
  padding: 0.8rem 1.2rem 0.8rem 1.2rem;
}

.c1 .react-datepicker-wrapper {
  max-width: 17rem;
}

.c2.c2.form-control.is-valid {
  background-image: none;
}

.c2.c2.form-control.is-invalid {
  border-color: #d8292f !important;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.c0 {
  margin-left: 1rem;
  font-size: 1.4rem;
  grid-template-columns: [controls] 1fr;
}

.c0 label {
  font-family: 'BCSans-Bold';
}

.c0 .check-field {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 .check-field label {
  width: 100%;
}

.c0 .form-group {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .form-group .small {
  width: 70%;
}

.c0 > .form-label {
  grid-column: controls;
  font-family: 'BcSans-Bold';
}

.c0 .form-control {
  font-family: 'BcSans';
}

.c0 .required .datepicker-label:after {
  content: ' *';
}

.c4 {
  margin-left: -1rem;
  padding-left: 1rem;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  border-radius: 0.5rem;
}

.c4 input {
  width: 100%;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

<div
    class="c0"
  >
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <div
          class="c1 required form-group"
        >
          <label
            class="form-label"
            for="datepicker-receivedDate"
          >
            <span
              class="datepicker-label"
            >
              Sent date:
            </span>
             
          </label>
          <div
            class="react-datepicker-wrapper d-block"
          >
            <div
              class="react-datepicker__input-container react-datepicker__view-calendar-icon"
            >
              <svg
                class="react-datepicker__calendar-icon"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M96 32V64H48C21.5 64 0 85.5 0 112v48H448V112c0-26.5-21.5-48-48-48H352V32c0-17.7-14.3-32-32-32s-32 14.3-32 32V64H160V32c0-17.7-14.3-32-32-32S96 14.3 96 32zM448 192H0V464c0 26.5 21.5 48 48 48H400c26.5 0 48-21.5 48-48V192z"
                />
              </svg>
              <input
                autocomplete="off"
                class="c2 form-control date-picker"
                id="datepicker-receivedDate"
                name="receivedDate"
                placeholder="MMM DD, YYYY"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <div
          class="form-group"
        >
          <label
            class="form-label"
            for="input-leasePaymentMethodType.id"
          >
            <span
              class="control-label"
            >
              Method:
            </span>
             
          </label>
          <select
            class="form-select form-control"
            id="input-leasePaymentMethodType.id"
            name="leasePaymentMethodType.id"
          >
            <option
              class="option"
              data-testid="select-option-CASH"
              value="CASH"
            >
              Cash
            </option>
            <option
              class="option"
              data-testid="select-option-CHEQ"
              value="CHEQ"
            >
              Cheque
            </option>
            <option
              class="option"
              data-testid="select-option-CRDR"
              value="CRDR"
            >
              Credit / Debit
            </option>
            <option
              class="option"
              data-testid="select-option-EFT"
              value="EFT"
            >
              EFT
            </option>
            <option
              class="option"
              data-testid="select-option-H120"
              value="H120"
            >
              H120
            </option>
            <option
              class="option"
              data-testid="select-option-OTHR"
              value="OTHR"
            >
              Other
            </option>
            <option
              class="option"
              data-testid="select-option-POST"
              value="POST"
            >
              Post-dated cheque
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <div
          class="form-group"
        >
          <label
            class="form-label"
            for="input-amountTotal"
          >
            Total received ($) 
          </label>
          <div
            class="input-tooltip-wrapper"
          >
            <input
              class="form-control input-number"
              id="input-amountTotal"
              inputmode="numeric"
              name="amountTotal"
              placeholder=""
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div
          class="c3 c4"
        >
          <div
            class="c5"
          >
            <span
              class="tooltip-icon"
              data-testid="tooltip-icon-actual-calculation-tooltip"
              id="actual-calculation-tooltip"
            >
              <svg
                class="tooltip-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                />
              </svg>
            </span>
          </div>
          <div
            class="c3"
          >
            <div
              class="form-group"
            >
              <label
                class="form-label"
                for="input-amountPreTax"
              >
                Amount (before tax) 
              </label>
              <div
                class="input-tooltip-wrapper"
              >
                <input
                  class="form-control input-number small"
                  id="input-amountPreTax"
                  inputmode="numeric"
                  name="amountPreTax"
                  placeholder=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="form-group"
            >
              <label
                class="form-label"
                for="input-amountGst"
              >
                GST ($) 
              </label>
              <div
                class="input-tooltip-wrapper"
              >
                <input
                  class="form-control input-number small"
                  id="input-amountGst"
                  inputmode="numeric"
                  name="amountGst"
                  placeholder=""
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div
          class="invalid-feedback"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`PaymentForm component > renders with data as expected 1`] = `
<DocumentFragment>
  <div
    class="Toastify"
  />
  <div />
  .c1 .react-datepicker__calendar-icon {
  width: 2.4rem;
  height: 3rem;
  margin-top: 0.5rem;
  margin-right: 1.2rem;
  right: 0;
  pointer-events: none;
}

.c1 .react-datepicker__view-calendar-icon input {
  padding: 0.8rem 1.2rem 0.8rem 1.2rem;
}

.c1 .react-datepicker-wrapper {
  max-width: 17rem;
}

.c2.c2.form-control.is-valid {
  background-image: none;
}

.c2.c2.form-control.is-invalid {
  border-color: #d8292f !important;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: nowrap;
  -ms-flex-wrap: nowrap;
  flex-wrap: nowrap;
}

.c0 {
  margin-left: 1rem;
  font-size: 1.4rem;
  grid-template-columns: [controls] 1fr;
}

.c0 label {
  font-family: 'BCSans-Bold';
}

.c0 .check-field {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c0 .check-field label {
  width: 100%;
}

.c0 .form-group {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.c0 .form-group .small {
  width: 70%;
}

.c0 > .form-label {
  grid-column: controls;
  font-family: 'BcSans-Bold';
}

.c0 .form-control {
  font-family: 'BcSans';
}

.c0 .required .datepicker-label:after {
  content: ' *';
}

.c4 {
  margin-left: -1rem;
  padding-left: 1rem;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  border-radius: 0.5rem;
}

.c4 input {
  width: 100%;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  width: 100%;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

<div
    class="c0"
  >
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <div
          class="c1 required form-group"
        >
          <label
            class="form-label"
            for="datepicker-receivedDate"
          >
            <span
              class="datepicker-label"
            >
              Sent date:
            </span>
             
          </label>
          <div
            class="react-datepicker-wrapper d-block"
          >
            <div
              class="react-datepicker__input-container react-datepicker__view-calendar-icon"
            >
              <svg
                class="react-datepicker__calendar-icon"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M96 32V64H48C21.5 64 0 85.5 0 112v48H448V112c0-26.5-21.5-48-48-48H352V32c0-17.7-14.3-32-32-32s-32 14.3-32 32V64H160V32c0-17.7-14.3-32-32-32S96 14.3 96 32zM448 192H0V464c0 26.5 21.5 48 48 48H400c26.5 0 48-21.5 48-48V192z"
                />
              </svg>
              <input
                autocomplete="off"
                class="c2 form-control date-picker"
                id="datepicker-receivedDate"
                name="receivedDate"
                placeholder="MMM DD, YYYY"
                required=""
                type="text"
                value=""
              />
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <div
          class="form-group"
        >
          <label
            class="form-label"
            for="input-leasePaymentMethodType.id"
          >
            <span
              class="control-label"
            >
              Method:
            </span>
             
          </label>
          <select
            class="form-select form-control"
            id="input-leasePaymentMethodType.id"
            name="leasePaymentMethodType.id"
          >
            <option
              class="option"
              data-testid="select-option-CASH"
              value="CASH"
            >
              Cash
            </option>
            <option
              class="option"
              data-testid="select-option-CHEQ"
              value="CHEQ"
            >
              Cheque
            </option>
            <option
              class="option"
              data-testid="select-option-CRDR"
              value="CRDR"
            >
              Credit / Debit
            </option>
            <option
              class="option"
              data-testid="select-option-EFT"
              value="EFT"
            >
              EFT
            </option>
            <option
              class="option"
              data-testid="select-option-H120"
              value="H120"
            >
              H120
            </option>
            <option
              class="option"
              data-testid="select-option-OTHR"
              value="OTHR"
            >
              Other
            </option>
            <option
              class="option"
              data-testid="select-option-POST"
              value="POST"
            >
              Post-dated cheque
            </option>
          </select>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col-md-6"
      >
        <div
          class="form-group"
        >
          <label
            class="form-label"
            for="input-amountTotal"
          >
            Total received ($) 
          </label>
          <div
            class="input-tooltip-wrapper"
          >
            <input
              class="form-control input-number"
              id="input-amountTotal"
              inputmode="numeric"
              name="amountTotal"
              placeholder=""
              type="text"
              value=""
            />
          </div>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div
          class="c3 c4"
        >
          <div
            class="c5"
          >
            <span
              class="tooltip-icon"
              data-testid="tooltip-icon-actual-calculation-tooltip"
              id="actual-calculation-tooltip"
            >
              <svg
                class="tooltip-icon"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 8C119.043 8 8 119.083 8 256c0 136.997 111.043 248 248 248s248-111.003 248-248C504 119.083 392.957 8 256 8zm0 110c23.196 0 42 18.804 42 42s-18.804 42-42 42-42-18.804-42-42 18.804-42 42-42zm56 254c0 6.627-5.373 12-12 12h-88c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h12v-64h-12c-6.627 0-12-5.373-12-12v-24c0-6.627 5.373-12 12-12h64c6.627 0 12 5.373 12 12v100h12c6.627 0 12 5.373 12 12v24z"
                />
              </svg>
            </span>
          </div>
          <div
            class="c3"
          >
            <div
              class="form-group"
            >
              <label
                class="form-label"
                for="input-amountPreTax"
              >
                Amount (before tax) 
              </label>
              <div
                class="input-tooltip-wrapper"
              >
                <input
                  class="form-control input-number small"
                  id="input-amountPreTax"
                  inputmode="numeric"
                  name="amountPreTax"
                  placeholder=""
                  type="text"
                  value=""
                />
              </div>
            </div>
            <div
              class="form-group"
            >
              <label
                class="form-label"
                for="input-amountGst"
              >
                GST ($) 
              </label>
              <div
                class="input-tooltip-wrapper"
              >
                <input
                  class="form-control input-number small"
                  id="input-amountGst"
                  inputmode="numeric"
                  name="amountGst"
                  placeholder=""
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="row"
    >
      <div
        class="col"
      >
        <div
          class="invalid-feedback"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
